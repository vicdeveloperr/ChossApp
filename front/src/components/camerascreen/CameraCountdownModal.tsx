import { View, Text, StyleSheet } from "react-native";
import { primaryColor } from "../../utils/colors";
import { useCountdownStore } from "../../stateManagement/";
import ScreenDarkModal from "../ScreenDarkModal";

const CameraCountdownModal: React.FC = () => {
  const { countdown } = useCountdownStore((state) => state);

  return (
    <ScreenDarkModal transparent>
      <View
        testID="CameraCountdownModal"
        style={styles.container}
      >
        <Text style={styles.timer}>{countdown}</Text>
      </View>
    </ScreenDarkModal>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
  },
  timer: {
    fontSize: 80,
    fontWeight: "bold",
    color: primaryColor,
  },
});

export default CameraCountdownModal;
