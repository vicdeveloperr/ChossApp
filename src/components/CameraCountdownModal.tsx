import { Modal, View, Text, StyleSheet } from "react-native";
import { useState, useEffect } from "react";
import { primaryColor } from "../styles/colors";

const CameraCountdownModal: React.FC = () => {
  const [countdown, setCountdown] = useState(3);

  useEffect(() => {
    let timer: NodeJS.Timeout;

    timer = setInterval(() => {
      setCountdown((prevCount) => (prevCount > 0 ? prevCount - 1 : 0));
    }, 1000);

    return () => {
      clearInterval(timer);
    };
  }, []);

  return (
    <Modal
      visible={true}
      animationType="slide"
      transparent
    >
      <View style={styles.container}>
        <Text style={styles.timer}>{countdown}</Text>
      </View>
    </Modal>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
  },
  timer: {
    fontSize: 80,
    fontWeight: "bold",
    color: primaryColor,
  },
});

export default CameraCountdownModal;
