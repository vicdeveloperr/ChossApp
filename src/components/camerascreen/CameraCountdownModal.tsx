import { View, Text, StyleSheet } from "react-native";
import { primaryColor } from "../../utils/colors";
import { useCountdownStore } from "../../stateManagement/stores";
import ScreenDarkModal from "../ScreenDarkModal";

const CameraCountdownModal: React.FC = () => {
  const { countdown } = useCountdownStore((state) => state);

  return (
    <ScreenDarkModal transparent>
      <View style={styles.container}>
        <Text style={styles.timer}>{countdown}</Text>
      </View>
    </ScreenDarkModal>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
  },
  timer: {
    fontSize: 80,
    fontWeight: "bold",
    color: primaryColor,
  },
});

export default CameraCountdownModal;
