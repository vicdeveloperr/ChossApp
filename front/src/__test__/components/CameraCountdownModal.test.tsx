import renderer from "react-test-renderer";
import CameraCountdownModal from "../../components/camerascreen/CameraCountdownModal";
import { render } from "@testing-library/react-native";

jest.mock("../../stateManagement/", () => ({
  useCountdownStore: jest.fn().mockReturnValue({ countdown: 3 }),
}));

describe("<CameraCountdownModal />", () => {
  it("Renderiza correctamente", () => {
    const tree = renderer.create(<CameraCountdownModal />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("Al renderizar, muestra el nÃºmero 3 en el primer instante", () => {
    const { getByText } = render(<CameraCountdownModal />);

    expect(getByText("3")).toBeTruthy();
  });
});
