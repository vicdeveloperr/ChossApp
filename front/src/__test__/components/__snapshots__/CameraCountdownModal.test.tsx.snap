// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CameraCountdownModal /> Renderiza correctamente 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "rgba(0,0,0,0.7)",
        "bottom": 0,
        "display": "flex",
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      },
      undefined,
    ]
  }
>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    transparent={true}
    visible={true}
  >
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
      testID="CameraCountdownModal"
    >
      <Text
        style={
          {
            "color": "#C74F14",
            "fontSize": 80,
            "fontWeight": "bold",
          }
        }
      >
        3
      </Text>
    </View>
  </Modal>
</View>
`;
